cmd/cloud-controller-manager/controller-manager.go

```diff
@@ -31,7 +31,7 @@ import (
 	// NOTE: Importing all in-tree cloud-providers is not required when
 	// implementing an out-of-tree cloud-provider.
 	_ "k8s.io/component-base/metrics/prometheus/clientgo" // load all the prometheus client-go plugins
-	_ "k8s.io/kubernetes/pkg/cloudprovider/providers"
+	_ "k8s.io/legacy-cloud-providers/providers"
```

Comment:
I actually don't think we want to move this to staging. Importing the legacy cloud providers is not desired behavior and we should discourage external consumers from doing that. The kubelet and kube-controller-manager imporint `pkg/cloudprovider/providers` is okay today because those will be removed anyways. For the cloud-controller-manager I think we can import the legacy providers individually.

Reply:
Will hand over this discussion to @cheftako

Reply:
As long as we are importing providers, you are importing all the in-tree providers. We need to break this link. For everything except the CCM, I believe it makes sense to have it in legacy-cloud-providers. They will all eventually delete this dependency, and as long as they have the dependency, it will be on all the providers under legacy-cloud-providers. For the CCM the correct dependency it per Provider. So the providers will have to over-ride it when they build their copy of the CCM. While we should make it easier to over-ride this dependency for the CCM I think it makes sense as we move things out.